src/dpp/cluster/emoji.cpp

```diff
@@ -42,4 +42,44 @@ void cluster::guild_emojis_get(snowflake guild_id, command_completion_event_t ca
 	rest_request_list<emoji>(this, API_PATH "/guilds", std::to_string(guild_id), "emojis", m_get, "", callback);
 }
 
+	//me.id.str()
+
+void cluster::application_emojis_get(command_completion_event_t callback) {
+	/* Because Discord can't be consistent, we can't just do `rest_request_list<emoji>` because all items are behind `items`.
+	 * so now we end up with this duplicating `rest_request_list` because we need to iterate the `items` array!!! Thanks Discord!!!!!!!
+	 */
+	post_rest(API_PATH "/application", me.id.str(), "emojis", m_get, "", [this, callback](json &j, const http_request_completion_t& http) {
+		std::unordered_map<snowflake, emoji> list;
+		confirmation_callback_t e(this, confirmation(), http);
+		const std::string key{"id"};
+		if (!e.is_error()) {
+			// No const for `fill_from_json`.
+			auto emojis_list = j["items"];
+			for (auto & curr_item : emojis_list) {
+				emoji t_emoji{};
+				list[snowflake_not_null(&curr_item, key.c_str())] = t_emoji.fill_from_json(&curr_item);
```

Comment:
can do this in one line without the temp var t_emoji:
```cpp
list[snowflake_not_null(&curr_item, key.c_str())] = emoji().fill_from_json(&curr_item);
```

Reply:
ah true, ty